src/lsqecc/simulation/logical_patch_state_simulation.py

```diff
@@ -307,16 +424,52 @@ def bring_active_operands_to_front(self, logical_op: llops.MultiBodyMeasurement)
         )
         involved_operand_idxs.sort()
 
-        swap_target_counter = 0
+        operand_swap_target_counter = 0
         operands_to_swap = deque(involved_operand_idxs)
         while len(operands_to_swap):
             front_of_queue = operands_to_swap.popleft()
-            self.logical_state.swap_operands(swap_target_counter, front_of_queue)
-            self.mapper.swap_patches(swap_target_counter, front_of_queue)
-            swap_target_counter += 1
+
+            self.logical_state.swap_operands(operand_swap_target_counter, front_of_queue)
+
+            # Update the patch->idx map
+            operand_uuid_list: List[List[uuid.UUID]] = []
+            for operand_idx, operand_size in enumerate(self.logical_state.get_operand_sizes()):
+                begin_operand = self.logical_state.get_idx_of_first_qubit_in_operand(operand_idx)
+                end_operand = begin_operand + operand_size  # TODO refactor into get_operator_bounds
+
+                operand_uuid_list.append(
+                    [self.mapper.get_uuid(idx) for idx in range(begin_operand, end_operand)]
+                )
+
+            operand_uuid_list[operand_swap_target_counter], operand_uuid_list[front_of_queue] = (
+                operand_uuid_list[front_of_queue],
+                operand_uuid_list[operand_swap_target_counter],
+            )
+            self.mapper.patch_location_to_logical_idx = dict(
+                [
+                    (patch_uuid, idx)
+                    for idx, patch_uuid in enumerate(itertools.chain(*operand_uuid_list))
+                ]
+            )
+
+            operand_swap_target_counter += 1
```

Comment:
Looks like I didn't make a separate commit for this change or lost it in rebase, but all this is doing is fixing a mismatch of indices when swapping operators